specutils/spectra/spectrum_mixin.py

```diff
@@ -9,6 +8,7 @@
 from astropy.wcs.wcsapi import HighLevelWCSWrapper
 
 from specutils.utils.wcs_utils import gwcs_from_array
+from ..utils.logger import log
```

Comment:
Better to use `logging.getLogger(__name__)` everywhere, this way it's possible to show the module when the message was emitted in the messages.

Reply:
Solid point, I erred on the side of being concise but I can deal with a few more lines for good reason.